File: sdks/python/applad/edge.py
"""Edge service."""


class Edge:
    def __init__(self, client):
        self.client = client

    def create(self, name: str, code: str, route: str = None, timeout: int = None):
        data = {"name": name, "code": code}
        if route is not None:
            data["route"] = route
        if timeout is not None:
            data["timeout"] = timeout
        return self.client._call("POST", "/edge/functions", data)

    def list(self):
        return self.client._call("GET", "/edge/functions")

    def get(self, function_id: str):
        return self.client._call("GET", f"/edge/functions/{function_id}")

    def update(self, function_id: str, **opts):
        return self.client._call("PUT", f"/edge/functions/{function_id}", opts)

    def delete(self, function_id: str):
        return self.client._call("DELETE", f"/edge/functions/{function_id}")

    def invoke(self, function_id: str, data: dict = None):
        return self.client._call("POST", f"/edge/functions/{function_id}/invoke", data or {})

    def list_executions(self, function_id: str):
        return self.client._call("GET", f"/edge/functions/{function_id}/executions")

File: sdks/dart/lib/src/server/edge.dart
import '../server_client.dart';

class Edge {
  final ApplAdServer _client;

  Edge(this._client);

  Future<Map<String, dynamic>> create({
    required String name,
    required String code,
    String? route,
    int? timeout,
  }) async {
    return _client.post('/v1/edge/functions', data: {
      'name': name,
      'code': code,
      if (route != null) 'route': route,
      if (timeout != null) 'timeout': timeout,
    });
  }

  Future<Map<String, dynamic>> list() async {
    return _client.get('/v1/edge/functions');
  }

  Future<Map<String, dynamic>> get(String functionId) async {
    return _client.get('/v1/edge/functions/$functionId');
  }

  Future<Map<String, dynamic>> update(
    String functionId, {
    String? name,
    String? code,
    String? route,
    int? timeout,
  }) async {
    return _client.put('/v1/edge/functions/$functionId', data: {
      if (name != null) 'name': name,
      if (code != null) 'code': code,
      if (route != null) 'route': route,
      if (timeout != null) 'timeout': timeout,
    });
  }

  Future<Map<String, dynamic>> delete(String functionId) async {
    return _client.delete('/v1/edge/functions/$functionId');
  }

  Future<Map<String, dynamic>> invoke(
    String functionId, {
    Map<String, dynamic>? data,
  }) async {
    return _client.post('/v1/edge/functions/$functionId/invoke',
        data: data ?? {});
  }

  Future<Map<String, dynamic>> listExecutions(String functionId) async {
    return _client.get('/v1/edge/functions/$functionId/executions');
  }
}
